### ISSUE TITLE
React 18.1 leading to a bug

### ISSUE THREAD

**Issue by artola** [konvajs/react-konva#693]
The following code prints 2 times "bug" in the console, and it should not happen.

```js
import { useEffect, useRef } from "react";
import { Layer, Stage } from "react-konva";

export default function App() {
  const layerRef = useRef(null);

  useEffect(() => {
    if (layerRef.current) {
      console.log("==> ok");
    } else {
      console.log("==> bug");
    }
  }, []);

  return (
    <div className="App">
      <Stage>
        <Layer ref={layerRef}></Layer>
      </Stage>
    </div>
  );
}
```

Deps:
- react-konva: 18.1.1
- konva: 7.2.5 (or higher makes no difference)
- react: 18.1.0
- react-dom: 18.1.0

Sandbox: https://codesandbox.io/s/react-18-1-bug-k8mejk?file=/src/App.js

The problem does not happen if I downgrade to `react-konva 18.1.0` or if I use the just released `react & react-dom 18.2.0`.

I must move to react 18.2  in such case, the problem is that we end up with 2 different versions `react-reconciler` and `scheduler` what does not seems the best.

An upgrade to support react 18.2 (just bumping the dependencies in react-konva) will be better, may be even the solution is just remove the `~0.` and use `^0.` will work for all.

**Comment by lavrton** [comment]
Double log is expected behavior for the last react in `StrictMode`: https://stackoverflow.com/a/61897567/512042

**Comment by artola** [comment]
@lavrton You are right at StrictMode behaviour, but I was trying to show that the effect is logging:

```
bug
bug
```

And it is because the ref is not properly attached in the moment that the effect is running.
This should only happen if someone forget to attach the ref to the element, but it is not the case in this code.

At the same time, doing the changes to the dependencies mentioned above, the problem disappear without changing the component at all.

Please re-open and evaluate alternative solution, other user will soon face similar problem.

**Comment by lavrton** [comment]
Sorry, you are right. I didn't understand your issue.

I just published `react-konva@18.2.1`. It should work correctly with `react@18.2.0`.

Please confirm.

**Comment by artola** [comment]
> Sorry, you are right. I didn't understand your issue.
> 
> I just published `react-konva@18.2.1`. It should work correctly with `react@18.2.0`.
> 
> Please confirm.

@lavrton I can confirm that `react-konva@18.2.1` works with `react@18.2.0.` and both use the same `scheduler@0.23.0` package.

Thanks a lot for your support.
